core/shared/src/main/scala/scalaz/zio/ZIO.scala

```diff
@@ -1245,6 +1245,28 @@ trait ZIOFunctions extends Serializable {
    */
   final def unsandbox[R >: LowerR, E <: UpperE, A](v: ZIO[R, Cause[E], A]): ZIO[R, E, A] = v.catchAll[R, E, A](halt)
 
+  /**
+   * Returns the identity effectful function, which performs no effects
+   */
+  final def identity[A >: LowerR]: ZIO[A, Nothing, A] = fromFunction[A, A](a => a)
+
+  /**
+   * Returns an effectful function that merely swaps the elements in a `Tuple2`.
+   */
+  final def swap[R >: LowerR, E <: UpperE,  A, B](implicit ev:  R =:= (A, B)): ZIO[R, E, (B, A)] = fromFunction[R, (B, A)](_.swap)
```

Comment:
I think you can generalize this to, `R <:< (A, B)`. It only needs to extend that type, not equal it. Ditto for the other ones.

It's a little contrived because of the bound on `R`, but there's not much we can do about that.